canon.go

```diff
@@ -91,3 +91,28 @@ func canonValue(chain *chain, in interface{}) (interface{}, bool) {
 
 	return out, true
 }
+
+func canonDecode(chain *chain, value interface{}, target interface{}) bool {
```

Comment:
* we should check if chain is already failed and return in this case
* we should check if target is nil and fail with AssertUsage in this case

Reply:
Ok,sure